src/Maker/MakeDockerDatabase.php

```diff
@@ -0,0 +1,154 @@
+     */
+    private $serviceVersion = 'latest';
+
+    public function __construct(FileManager $fileManager)
+    {
+        $this->fileManager = $fileManager;
+        $this->composeFilePath = sprintf('%s/docker-compose.yaml', $this->fileManager->getRootDirectory());
+    }
+
+    public static function getCommandName(): string
+    {
+        return 'make:docker:database';
+    }
+
+    public function configureCommand(Command $command, InputConfiguration $inputConfig): void
+    {
+        $command
+            ->setDescription('Adds a database container to your docker-compose.yaml file.')
+        ;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        $io->section('- Docker Compose Setup-');
+
+        $composeFileContents = '';
+        $statusMessage = 'Existing docker-compose.yaml not found: a new one will be generated!';
+
+        if ($this->fileManager->fileExists($this->composeFilePath)) {
+            $composeFileContents = $this->fileManager->getFileContents($this->composeFilePath);
+
+            $statusMessage = 'We found your existing docker-compose.yaml: Let\'s update it!';
+        }
+
+        $io->text($statusMessage);
+
+        $this->composeFileManipulator = new ComposeFileManipulator($composeFileContents);
+
+        $io->newLine();
+
+        $this->databaseChoice = strtolower($io->choice(
+            'Which database service will you be creating?',
+            ['MySQL', 'MariaDB', 'Postgres']
+        ));
+
+        $io->text([sprintf(
+            'For a list of supported versions, check out https://hub.docker.com/_/%s',
+            $this->databaseChoice
+        )]);
+
+        $this->serviceVersion = $io->ask('What version would you like to use?', DockerDatabaseServices::getSuggestedServiceVersion($this->databaseChoice));
+
+        if ($this->composeFileManipulator->serviceExists($this->serviceName)) {
+            $io->comment(sprintf('A <fg=yellow>"%s"</> service is already defined.', $this->serviceName));
+            $io->newLine();
+
+            $envTest[] = 'If you are using the Symfony Binary, it will expose the env_var associated with this service.';
+            $envTest[] = 'If you name the service \'database\', the binary will look for DATABASE_URL';
+
+            $io->text($envTest);
+
+            $this->serviceName = $io->ask(sprintf('What name should we call the new %s service? e.g. database', $this->serviceName));
+        }
+    }
+
+    public function generate(InputInterface $input, ConsoleStyle $io, Generator $generator): void
+    {
+        $io->newLine();
+
+        $service = DockerDatabaseServices::getDatabaseSkeleton($this->databaseChoice, $this->serviceVersion);
+
+        $this->composeFileManipulator->addDockerService($this->serviceName, $service);
+
+        $generator->dumpFile($this->composeFilePath, Yaml::dump($this->composeFileManipulator->getComposeData(), 20));
```

Comment:
Wait a sec... this isn't quite right. We're never using the `YamlSourceManipulator`. I mean, we use it for a hot second in ComposeFileManipulator to load the data, but then we do *not* actually use that class to update the contents. We need to do that - instead of using `Yaml::dump()` - else we're going to blow away any existing comments and formatting.

Reply:
done